benchmarks/distributed/rpc/ps/metrics/MetricsLogger.py

```diff
@@ -0,0 +1,92 @@
+from .CPUMetric import CPUMetric
+from .CUDAMetric import CUDAMetric
+
+
+class MetricsLogger:
+
+    CUDA = "cuda"
+    CPU = "cpu"
+
+    def __init__(self, rank, metric_class, overwrite_metrics=False):
+        self.rank = rank
+        self.metric_class = metric_class.lower()
+        self.overwrite_metrics = overwrite_metrics
+        self.metrics = {}
+
+    def add_metric(self, metric_type, key, metric_name):
+        if metric_type in self.metrics and key in self.metrics[metric_type] and not self.overwrite_metrics:
+            raise Exception("metric_type={} with key={} already exists".format(metric_type, key))
+
+        if metric_type not in self.metrics:
+            self.metrics[metric_type] = {}
+
+        if self.metric_class == self.CPU:
+            self.metrics[metric_type][key] = CPUMetric(metric_name)
+        elif self.metric_class == self.CUDA:
+            self.metrics[metric_type][key] = CUDAMetric(metric_name)
+
+        self.metrics[metric_type][key].record_start(self.rank)
+
+    def add_metric_end(self, metric_type, key):
+
+        if metric_type not in self.metrics or key not in self.metrics[metric_type]:
+            raise Exception("metric_type={} with key={} not found".format(metric_type, key))
+
+        if self.metrics[metric_type][key].get_end() is not None and not self.overwrite_metrics:
+            raise Exception("end for metric_type={} with key={} already exists".format(metric_type, key))
+
+        self.metrics[metric_type][key].record_end(self.rank)
+
+    def clear_metrics(self):
+        self.metrics.clear()
+
+    def get_metrics(self):
+        return self.metrics
+
+    r"""Method that processes the Metrics. 
+    It returns a dictionary containing keys 
+    that are the metric_type and metric_name combined. 
+    Each key has a list of elapsed times.
+
+     Args:
+        self: the instance of the class
```

Comment:
API doc usually don't include self, because users would use it as `instance. get_processed_metrics()`.

Reply:
okay.